### PR TITLE
Added holograms to TileMasterRitualStone and TileAltar

### PR DESCRIPTION
**Pull request by Arcaratus**
-Right click with either Ritual Diviner or (name pending) Sanguine Sanctum to show up the hologram
-Ritual hologram disappears once ritual is activated
-Altar hologram disappears once altar reaches specified tier
-Fixed the Sigil of Holding configs
-Someone still needs to add in a recipe for the Sigil of Holding
-Made the Blood Altar check the Sigil of Holding for IAltarManipulator and IAltarReader
-Disabled the Sanguine Sanctum right-click effect for now
-Kept the hologram from holding the Ritual Diviner in hand
-Someone needs to fix the lighting for the ritual hologram!

By the way, am I doing this branch thing right?

### CODE REVIEW

File: src/main/java/WayofTime/bloodmagic/altar/BloodAltar.java [hunk 1]
```diff
@@ -717,6 +724,20 @@ public int getBufferCapacity()
         return bufferCapacity;
     }
 
+    public EnumAltarTier getCurrentTierDisplayed()
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/altar/BloodAltar.java]: `@Getter` on `currentTierDisplayed`

File: src/main/java/WayofTime/bloodmagic/client/render/RenderMasterRitualStone.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.lwjgl.opengl.GL11;
+
+public class RenderMasterRitualStone extends TileEntitySpecialRenderer<TileMasterRitualStone>
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/client/render/RenderMasterRitualStone.java]: I would much rather not use a TESR for this. It's completely unneeded.

File: src/main/java/WayofTime/bloodmagic/item/ItemRitualDiviner.java [hunk 3]
```diff
@@ -220,27 +240,27 @@ public void addInformation(ItemStack stack, EntityPlayer player, List<String> to
                 {
                     switch (component.getRuneType())
                     {
-                    case BLANK:
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/ItemRitualDiviner.java:223]: Unneeded formatting change.

File: src/main/java/WayofTime/bloodmagic/item/ItemRitualDiviner.java [hunk 4]
```diff
@@ -335,20 +355,20 @@ public void cycleDirection(ItemStack stack, EntityPlayer player)
         EnumFacing newDirection;
         switch (direction)
         {
-        case NORTH:
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/ItemRitualDiviner.java:338]: Unneeded formatting change.

File: src/main/java/WayofTime/bloodmagic/item/ItemRitualDiviner.java [hunk 5]
```diff
@@ -552,16 +572,16 @@ public boolean canPlaceRitualStone(EnumRuneType rune, ItemStack stack)
         int meta = stack.getItemDamage();
         switch (rune)
         {
-        case BLANK:
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/ItemRitualDiviner.java:555]: Unneeded formatting change.

File: src/main/java/WayofTime/bloodmagic/item/ItemSanguineBook.java [hunk 6]
```diff
@@ -39,27 +47,75 @@ public EnumActionResult onItemUse(ItemStack stack, EntityPlayer player, World wo
         if (world.isRemote)
             return super.onItemUse(stack, player, world, pos, hand, facing, hitX, hitY, hitZ);
 
-        IBlockState hitState = world.getBlockState(pos);
-        if (player.isSneaking() && hitState.getBlock() instanceof IDocumentedBlock)
+//        IBlockState hitState = world.getBlockState(pos);
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/ItemSanguineBook.java]: Really?

File: src/main/java/WayofTime/bloodmagic/tile/TileMasterRitualStone.java [hunk 7]
```diff
@@ -220,6 +234,40 @@ public void stopRitual(Ritual.BreakType breakType)
         }
     }
 
+    public Ritual getCurrentDisplayedRitual()
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/tile/TileMasterRitualStone.java]: `@Getter` on `currentDisplayedRitual`

File: src/main/java/WayofTime/bloodmagic/tile/TileMasterRitualStone.java [hunk 8]
```diff
@@ -220,6 +234,40 @@ public void stopRitual(Ritual.BreakType breakType)
...
+    {
+        if (currentDisplayedRitual != null && currentDisplayedRitual.getName().equals(ritual.getName()))
+            return false;
+        else
+            this.currentDisplayedRitual = ritual;
+
+        return true;
+    }
+
+    public EnumFacing getCurrentDisplayedDirection()
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/tile/TileMasterRitualStone.java]: `@Getter` on `currentDisplayedDirection`

File: src/main/java/WayofTime/bloodmagic/util/handler/event/ClientHandler.java [hunk 9]
```diff
@@ -236,27 +252,27 @@ private void renderRitualStones(EntityPlayerSP player, float partialTicks)
 
                 switch (ritualComponent.getRuneType())
                 {
-                case BLANK:
```
Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/util/handler/event/ClientHandler.java:239]: Unneeded formatting change.

File: src/main/java/WayofTime/bloodmagic/client/render/RenderMasterRitualStone.java (hunk 2 above)
Comment (Arcaratus) [inline src/main/java/WayofTime/bloodmagic/client/render/RenderMasterRitualStone.java]: What should I be using?

Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/client/render/RenderMasterRitualStone.java]: The same thing you used for the ritual rendering before? The data doesn't need to be saved serverside and/or synced so I'm actually not sure why you're using fields in the Tile's themselves either.

File: src/main/java/WayofTime/bloodmagic/item/ItemSanguineBook.java (hunk 6 above)
Comment (Arcaratus) [inline src/main/java/WayofTime/bloodmagic/item/ItemSanguineBook.java]: ?

Comment (TehNut) [inline src/main/java/WayofTime/bloodmagic/item/ItemSanguineBook.java]: You commented out an entire feature

**Comment by Arcaratus** [comment]
https://www.youtube.com/watch?v=1alHA1n3Ojo

**Comment by TehNut** [comment]
Does the server run correctly?

**Comment by Arcaratus** [comment]
Tested and works!

**Comment by WayofTime** [comment]
So, the Book of Sanguine I felt was going to be useful for things other than just the Altar - is there a way that we can set the tier of the altar hologram without "hard coding" that function, so to speak?

Edit: Also, are we still going to add the ability for the book to display information about the blocks? Such as its original purpose, which would tell you what blocks are missing and where?

**Comment by Arcaratus** [comment]
That ability is still in, about the hard coding though, not sure about that. Up to you to decide, or maybe toss it to some users and hear some reviews

**Comment by TehNut** [comment]
Cycle through altar tiers when you right click on it

**Comment by TehNut** [comment]
Why did you move the documentation feature to right click instead of use?

File: src/main/java/WayofTime/bloodmagic/block/BlockAltar.java [hunk 10]
```diff
@@ -152,6 +153,16 @@ public boolean onBlockActivated(World world, BlockPos pos, IBlockState state, En
                 playerItem.getItem().onItemRightClick(playerItem, world, player, hand);
                 return true;
             }
+
+            if (playerItem.getItem() instanceof ItemSigilHolding)
```
Comment (WayofTime) [inline src/main/java/WayofTime/bloodmagic/block/BlockAltar.java]: Could ItemSigilHolding implement IAltarReader?

**Comment by WayofTime** [comment]
Other than that, first read through seems good.

File: src/main/java/WayofTime/bloodmagic/block/BlockAltar.java (hunk 10 above)
Comment (Arcaratus) [inline src/main/java/WayofTime/bloodmagic/block/BlockAltar.java]: I could try that and see if it works

Comment (Arcaratus) [inline src/main/java/WayofTime/bloodmagic/block/BlockAltar.java]: My bad, the Divination Sigil wasn't bound when I tested

**Pull request merged**